crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -191,36 +196,33 @@ impl Inner {
             .with_context(|| format!("no receipt {hash:?}"))?;
         let gas_used = receipt.gas;
         let effective_gas_price = receipt.effective_gas_price;
-        let auction_external_prices = self
-            .persistence
-            .auction_prices(auction_id)
-            .await
-            .with_context(|| {
-                format!("no external prices for auction id {auction_id:?} and tx {hash:?}")
-            })?
-            .into_iter()
-            .map(|(token, price)| (token.0, price.get().into()))
-            .collect::<BTreeMap<_, _>>();
+        let auction = Postgres::find_competition(auction_id, ex)
+            .await?
+            .context(format!(
+                "missing competition for auction_id={:?}",
+                auction_id
+            ))?
+            .common
+            .auction;
```

Comment:
Didn't we just create this new function in #2702? Is it really the intention to remove it again? Since this function is not used anywhere else, should we properly remove it?

Reply:
Not sure I understand why is this needed? Is it because during autopilot competition we only store auction prices related to tokens contained in the user orders from winning solution? But we actually need prices for all tokens?
If that is true, than I would rather save all the prices to auction_prices (this is how it was implemented initially but I was playing smart and wanted to optimize for db space and save only what we need ) and keep the https://github.com/cowprotocol/services/pull/2702 function. We are not supposed to read from SolverCompetition table.